schema.ts

```diff
@@ -696,6 +703,16 @@ export type SchemaType<T extends Schema> = T extends BooleanSchema
 
 export type ValidTypes = boolean | number | string | object | boolean[] | number[] | string[] | object[];
 
+/**
+ * Utility that examines a JavaScript value and attempts to infer a schema definition
+ * that describes it.
+ *
+ * It is vastly preferable to define a schema manually. A clear and accurate schema is one of the
+ * fundamentals of a good pack. However, for data that is truly dynamic for which a schema can't
+ * be known in advance nor can a function be written to generate a dynamic schema from other
+ * inputs, it may be useful to us this helper to sniff the return value and generate a basic
+ * inferred schema from it.
```

Comment:
Are there any limitations on the object passed in? Can it contain arrays of sub-objects, for instance? How does select the ID and primary?

Reply:
I don't think there's any limitations, it works recursively so should handle nested objects just fine.

It doesn't infer an `id` or `primary`, so I added a sentence about that.